lib/ts/recipe/session/sessionClass.ts

```diff
@@ -164,4 +145,82 @@ export default class Session implements SessionContainerInterface {
         }
         return sessionInfo.expiry;
     };
+
+    assertClaims = async (claimValidators: SessionClaimValidator[], userContext?: any): Promise<void> => {
```

Comment:
This function should call the recipeImplementation function which should have all the logic for actually doing the checking and throwing - not the other way around.